kbcstorage/base.py

```diff
@@ -0,0 +1,139 @@
+"""
+Base classes for constructing the client.
+
+Primarily exposes a base Endpoint class which deduplicates functionality across
+various endpoints, such as tables, workspaces, jobs, etc. as described in the
+`Storage API documentation`.
+
+
+.. _Storage API documentation:
+    http://docs.keboola.apiary.io/
+"""
+from urllib.parse import urljoin
+
+import requests
+
+
+class Endpoint:
+    """
+    Base class for implementing a single endpoint related to a single entities
+    as described in the Storage API.
+
+    Attributes:
+        path (str): URL for this endpoint.
+        token (str): A key for the Storage API.
+    """
+    def __init__(self, root, extension, token):
+        """
+        Create an endpoint.
+
+        Args
+            root (str): Root url of API. eg.
+                "https://connection.keboola.com/v2/storage/"
```

Comment:
In my mind the client constructs the string and passes it to the Endpoint. ie.

```python
class Client:
    def __init__(self, base_url, token):
        api_version = 'v2/storage'
        root_url = base_url + '/' + api_version
        self.buckets = Buckets(root_url, token)
        ...
```

Does that work? Should API version also be a parameter? Or do you think each endpoint should construct the API string?